### DISCUSSION TITLE
Image resolution and illumination requirements for port processing?

### DISCUSSION THREAD

**Discussion by hardev2z** [openPupil/Open-PupilEXT#29] [General]
Hi all, 

I am interested in using the PupilEXT open source software package using a high-res camera installed on the dashboard of a vehicle and pointing towards the driver’s face. We will use the package in post-processing mode, so no live data will be fed to the algorithm. As only part of the captured image will contain the eye(s), we should ensure that the resolution of this partially captured image is still sufficient in resolution. Are there any minimum requirements in resolution for the algorithm to work properly? And what about illumination conditions, as these typically change in a vehicle? The camera we are considering has both RGB and IR spectrum capturing, providing an IR light source. Does the PupilEXT software provide the possibility to filter out only the RGB or IR spectrum? 

Many thanks if you’d be able to sketch some of these in a couple of brief answers. It is much appreciated.

**Comment by BZandi** [comment]
**Resolution:** There should be no minimum requirements in terms of resolution for the images. With lower resolution, you impact the accuracy of the pupil detection, of course. It is advised to test the performance of the algorithms on test images before conducting the experiment.

**Lighting conditions:** Usually, an IR light source is used and a NIR colour filter is applied in front of the camera's lens optic to block the visible spectrum range below ~780nm as far as possible, leading to less reflections in the eye's cornea and better pupil detection. See also this discussion topic ([https://github.com/openPupil/Open-PupilEXT/discussions/5](https://github.com/openPupil/Open-PupilEXT/discussions/5)). Our test setup is also described in detail in our published paper ([https://www.frontiersin.org/articles/10.3389/fnins.2021.676220/abstract](https://www.frontiersin.org/articles/10.3389/fnins.2021.676220/abstract))

**RGB:** In our setup, we used only monochrome cameras from Basler. I don't know whether an RGB camera works with PupilEXT. As far as I remember, we cast the RGB Images in monochrome before feeding the images into the algorithms. You need to test this.

Note that we also provide a python version of PupilEXT where you have more freedom of degree in processing the images and controlling your RGB cameras. See this repository: [https://github.com/openPupil/PyPupilEXT](https://github.com/openPupil/PyPupilEXT)

**Reply by hardev2z** [reply]
Hi Mr Zandi, 

thank you for the elaborate and clear answer. Once I have some concrete results, I can let you and the community know how it went. 

Regards, Harald
